### PR TITLE
Add all_tenants configration to keystone

### PR DESCRIPTION
**Pull request by ghost**
`all_tenants` specify authority for all tenants.
If both tenants and all_tenants are setting, tenants is preferred.

Suppose existing tenants are foo, bar and baz, this method work as following:

#### keystone.yml

```
suzupy:
  email: suzupy@example.com
  all_tenants: member
```

#### result

```
suzupy:
  email: suzupy@example.com
  tenants:
    foo: member
    bar: member
    baz: member
```

#### keystone.yml

```
suzupy:
  email: suzupy@example.com
  all_tenants: member
  tenants:
    foo: admin
```

#### result

```
suzupy:
  email: suzupy@example.com
  tenants:
    foo: admin
    bar: member
    baz: member
```

### CODE REVIEW

File: lib/kaname/resource.rb [hunk 1]
```diff
@@ -4,11 +4,25 @@ module Kaname
...
-        @_yaml = if File.exists?(filename)
-                   YAML.load_file(filename)
-                 else
-                   nil
-                 end
+        if File.exists?(filename)
+          @_yaml = YAML.load_file(filename)
+          expand_all_tenants
+        else
+          nil
```
Comment (pyama86) [inline lib/kaname/resource.rb]: I think that it is unnecessary return value.
```ruby
a =  if false
    puts "false"
end
puts a.nil?
# => true
```

File: lib/kaname/resource.rb [hunk 2]
```diff
@@ -4,11 +4,25 @@ module Kaname
...
     class << self
       def yaml(filename = 'keystone.yml')
-        @_yaml = if File.exists?(filename)
-                   YAML.load_file(filename)
-                 else
-                   nil
-                 end
+        if File.exists?(filename)
+          @_yaml = YAML.load_file(filename)
+          expand_all_tenants
```
Comment (pyama86) [inline lib/kaname/resource.rb]: It seems good to cache the return value of expand_all_tenants
```ruby
@_yaml ||= expand_all_tenants(YAML.load_file(filename))
```

File: lib/kaname/resource.rb [hunk 3]
```diff
@@ -5,17 +5,16 @@ class Resource
...
-          @_yaml = YAML.load_file(filename)
-          expand_all_tenants
+          @_yaml ||= expand_all_tenants(YAML.load_file(filename))
         end
       end
 
       private
 
-      def expand_all_tenants
+      def expand_all_tenants _yaml
```
Comment (pyama86) [inline lib/kaname/resource.rb]: What kind of policy does the underscore use?
I think that a commonly used example is the name of a private method etc.

Comment (ghost) [reply lib/kaname/resource.rb]: This is differentiation between `yaml  variable` and `yaml method`.
@_yaml is used under this policy, isn't it?

Comment (pyama86) [reply lib/kaname/resource.rb]: oh...ok.

**Pull request merged**